Ticket #2869 Refactor experiment page

olgamelnichuk opened the pull request:
Experiment chart data moved out from the ExperimentResultAdapter.
@alf239 — please review

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- forgot to remove obsoleted code

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- efNames and efCuratedNames are combined into single object

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- operation priority fixed

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- DoubleIndexIterator fix: Collection -> Iterable

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- DoubleIndexIterator moved to atlas-utils

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- DoubleIndexIterator: test methods are renamed

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- FactorValueComparator moved to the utils module; added a unit test;

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- added a Note about using guava's multimaps approach in the ExperimentPlot

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- moved out BoxAndWhisker class from ExperimentPlot; added the unit test;

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- removed explicit cast

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- removed creation date from class JavaDoc

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- removed unused code

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- UpDownExpression class moved to the atlas-model

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- removed date from the JavaDoc of the class

alf239 commented on atlas-utils/src/main/java/uk/ac/ebi/gxa/utils/FactorValueComparator.java:
```diff
@@ -0,0 +1,77 @@
+ * Compares two strings by numeric prefix or suffix.
+ * <p/>
+ * While
+ * - An empty string is considered grater than non-empty one.
```
a typo

alf239 commented on atlas-utils/src/main/java/uk/ac/ebi/gxa/utils/DoubleIndexIterator.java:
```diff
@@ -0,0 +1,95 @@
+import java.util.Iterator;
+
+/**
+ * @author Olga Melnichuk
```
proper javadoc would be helpful: it's not at all clear what exactly the class does.

Something along the lines of, "Iterator over the list of lists allowing client to get current indices, which is helpful in case of parallel arrays" would be helpful

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- removed  accidentally committed code

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- optimized imports

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- UpDownExpression and UpDownCondition added to the atlas-module

Commit:
Merge remote branch 'origin/master' into trac2869

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- reference to the trac ticket added

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- added FloatMatrixProxy as workaround of unnecessary copying matrix read from NetCDF

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- a typo

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- java Doc added to DoubleListIterator util

Commit:
Unnecessary return value — no chaining here so far

alf239 commented on atlas-utils/src/main/java/uk/ac/ebi/gxa/utils/FactorValueComparator.java:
```diff
@@ -0,0 +1,77 @@
+ *
+ * @author Olga Melnichuk
+ */
+public class FactorValueComparator implements Comparator<String> {
```
Comparator doesn't implement Serializable

This class implements the Comparator interface. You should consider whether or not it should also implement the Serializable interface. If a comparator is used to construct an ordered collection such as a TreeMap, then the TreeMap will be serializable only if the comparator is also serializable. As most comparators have little or no state, making them serializable is generally easy and good defensive programming.

alf239 replied:
fixed

Commit:
Setting up findbugs ignore pattern, fixing what's fixable

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
-FactorValueComparator is serializable now

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- added exclude file to findbugs-maven-plugin

Commit:
Removing unnecessary parentheses

Commit:
A bit more concise code

Commit:
Fixed seed in order to make tests reproducible

Commit:
Code clarified, added notes on test failures

alf239 commented on atlas-web/src/main/java/ae3/model/AtlasGene.java:
```diff
@@ -402,9 +405,9 @@ public class AtlasGene {
             // Retrieve all up/down counts and pvals/tStatRanks
             for (ExperimentInfo exp : allExperimentsForAttribute) {
-                boolean isNo = ExpressionAnalysis.isNo(exp.getpValTStatRank().getPValue(), exp.getpValTStatRank().getTStatRank());
-                if (!isNo) {
```
It was !isNo — transformed to isUpOrDown.

Shouldn't it be !isNonDe instead?

alf239 replied:
It should actually be UP or DOWN - so the new code is, well, "better"

alf239 commented on atlas-web/src/test/java/uk/ac/ebi/gxa/plot/BoxAndWhiskerTest.java:
```diff
@@ -0,0 +1,75 @@
+    private static BoxAndWhisker newBoxAndWhisker(Float... data) {
+        List<Float> list = Arrays.asList(data);
+        Collections.shuffle(list);
+        return new BoxAndWhisker(list, null);
```
Please set up seed for this test - we need reproducibility

Commit:
Ticket #2869 Refactor experiment page
Experiment chart data moved out from the ExperimentResultAdapter
- added random seed to reproduce failed tests

alf239 commented:
Reviewed, OK.

Created trac tickets:
#2910 NetCDF missing value conflicts
#2943 Make NetCDFReader work through NetCDFProxy
#2944 StatisticsType vs UpDownCondition

@ostolop - please review/prioritise the tickets
